ext/date/tests/bug73239.phpt

```diff
@@ -0,0 +1,14 @@
+--TEST--
+Bug #73239 (Odd warning/exception message with invalid timezones)
+--FILE--
+<?php
+ini_set('date.timezone', 'dummy');
+try {
+	$dt = new DateTime('now');
+} catch (Exception $e) {
+	echo $e->getMessage(), "\n";
+}
+?>
+--EXPECTF--
+Warning: ini_set(): Invalid date.timezone value 'dummy' in %sbug73239.php on line %d
+DateTime::__construct(): Invalid date.timezone value 'dummy', we selected the timezone 'UTC' for now.
```

Comment:
I think you might have mixed things up? The "we selected the timezone 'UTC' for now" should be on the ini_set() error message (where it makes sense) and not the DateTime::__construct() message (where it doesn't, because of the exception).

Reply:
The patch, as is, removes both "we selected the timezone 'UTC' for now" messages; only that test expection isn't in line with that. I agree though, that the message raised from `PHP_INI_MH(OnUpdate_date_timezone)` might better stay as is (well, not really the wording; I'd go with passive voice instead).

Reply:
No, I did not mix this up. The `ini_set()` warning is correct. At that state, PHP never fell back to `UTC`, it just shown the warning to hint that you messed up.

The warning for **DateTime** was correct as in the test, the code is wrong. It should very much tell you there that there was an actual fallback to `UTC`. And I have updated the wording.

I don't want to change the behaviour here.

Reply:
Well, but throwing an exception with the message "DateTime::__construct(): UTC was used as timezone, because the date.timezone value 'dummy' is invalid" doesn't make sense. If UTC *was* used, there is no reason to throw an exception. And when an exception is thrown, that UTC is irrelevant. Or do I miss something?

Reply:
Oh yes, of course. That's why I originally had removed that part of the sentence. I'll fix that back then. Need more :tea: .

Reply:
It isn't actually that simple. The location where this warning message is thrown, is in `guess_timezone`:

```
        if (!timelib_timezone_id_is_valid(DATEG(default_timezone), tzdb)) {
            php_error_docref(NULL, E_WARNING, "UTC was used as timezone, because the date.timezone value '%s' is invalid", DATEG(default_timezone));
            return "UTC";
        }
```
Where the warning *is* correct. This same `guess_timezone` function is also called in `date_initialize` (via `get_timezone_info()`). `get_timezone_info()` is used plenty of times, where the warning message should be shown. However, `get_timezone_info()` is also called as part of the `DateTime` constructor, where then the exception message is indeed silly.